### PR TITLE
Make test suite runnable on iOS

### PR DESCRIPTION
**Pull request by lugehorsam**
- Change our `test` cmake project to bring in `nakama-sdk` as a dependency, not the other way around. This was necessary so we could embed `nakama-sdk` as a framework into the iOS application bundle.
- [x] Modify CI scripts to point to the new test executable location

### CODE REVIEW

File: .github/actions/cmake-build/action.yml [hunk 1]
```diff
@@ -80,6 +80,14 @@ runs:
       shell: bash
     - run: cmake --install ./build/${{ inputs.preset }} --config ${{ inputs.build_type }}
       shell: bash
+    - run: cd test && cmake --preset ${{ inputs.preset }}
+      shell: bash
+    - run: cd test && cmake --preset ${{ inputs.preset }}
```
Comment (redbaron) [inline .github/actions/cmake-build/action.yml:85]: duplicate?

File: .github/workflows/pr.yml [hunk 2]
```diff
@@ -21,13 +21,13 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       - shell: bash
         run: |
-          ./build/win-x64/test/MinSizeRel/nakama-test || {
+          ./out/test/win-x64/nakama-test.exe || {
```
Comment (redbaron) [inline .github/workflows/pr.yml:24]: I think packaging scripts currently expect everything in "out" to be a release artifacts

Comment (lugehorsam) [reply .github/workflows/pr.yml]: @redbaron yep this was actually a misconfigured path (fixed to `out/test`)

**Comment by lugehorsam** [comment]
@redbaron ready for re-review...let's see how these builds go.

**Review by redbaron — APPROVED** [review]

**Pull request merged**